packages/flutter_tools/templates/app_shared/windows.tmpl/runner/utils.cpp

```diff
@@ -45,17 +45,18 @@ std::string Utf8FromUtf16(const wchar_t* utf16_string) {
   if (utf16_string == nullptr) {
     return std::string();
   }
+  int source_length = static_cast<int>(wcslen(utf16_string));
   int target_length = ::WideCharToMultiByte(
       CP_UTF8, WC_ERR_INVALID_CHARS, utf16_string,
-      -1, nullptr, 0, nullptr, nullptr);
+      source_length, nullptr, 0, nullptr, nullptr);
   std::string utf8_string;
   if (target_length == 0 || target_length > utf8_string.max_size()) {
     return utf8_string;
   }
   utf8_string.resize(target_length);
```

Comment:
Should this be `source_length`?

Reply:
Do you mean the if (target_length == 0...) condition? I think we want to check the resulting length (target_length) which will be 0 e.g. if there are invalid characters in an non-empty utf16_string (note WC_ERR_INVALID_CHARS is set).

Reply:
`target_length` is write -- WideCharToMultibyte returns the number of characters that are written to the `lpMultiByteStr` out param or, in the case where we pass null for that param, the number that are required. We then resize `utf8_string` (the target buffer) to that size and do the write.

Reply:
Hmm, isn't std::string devoid of null terminating char? If we inlude the C string null char in the resulting std::string, it will become an zero UTF-8 code point.

Reply:
Ah I see - the issue is that we're actively including the nul-terminator as part of the string contents due to the length passed in and thus generating a C++ output string whose size is therefore one greater than it should be because of the extra nul-terminator byte. This makes sense. Apologies for the delay in getting back to this.